flutter_secure_storage/android/src/main/java/com/it_nomads/fluttersecurestorage/FlutterSecureStoragePlugin.java

```diff
@@ -21,30 +22,30 @@ public class FlutterSecureStoragePlugin implements MethodCallHandler, FlutterPlu
 
     private MethodChannel channel;
     private FlutterSecureStorage secureStorage;
-    private HandlerThread workerThread;
-    private Handler workerThreadHandler;
+    private ExecutorService executor;
     private FlutterPluginBinding binding;
 
     @Override
     public void onAttachedToEngine(FlutterPluginBinding binding) {
         this.binding = binding;
-        workerThread = new HandlerThread("fluttersecurestorage.worker");
-        workerThread.start();
-        workerThreadHandler = new Handler(workerThread.getLooper());
+        int availableCores = Runtime.getRuntime().availableProcessors();
+        int threadCount = Math.min(4, Math.max(2, availableCores));
+        this.executor = Executors.newFixedThreadPool(threadCount);
```

Comment:
Why use a fixed thread, when we can also use newCachedThreadPool which in my opinion seems more useful for these i/o actions? This also removes the need for looking up cores and threads.